aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp

```diff
@@ -296,18 +296,26 @@ Tensor flash_attention_helper(
     int64_t Nnz_q{cumulative_sequence_length_q[-1].item<int64_t>()};
 
     // For the packed case we need to set the output size for dim 2 to 1
-    auto atten_size = get_nested_size_tensor(query);
+    auto atten_size = get_nested_size_tensor(query).clone();
     atten_size.index({at::indexing::Slice(), 1}) = 1;
 
     auto qkv_buffer_reshaped =
-        get_buffer(query).view({Nnz_q, 3, num_heads, head_dim});
+        get_buffer(query).view({Nnz_q, 3, num_heads, head_dim}).transpose(0, 1).contiguous();
+
+    auto i0 = qkv_buffer_reshaped[0];
+    auto i1 = qkv_buffer_reshaped[1];
+    auto i2 = qkv_buffer_reshaped[2];
+
+    TORCH_CHECK(i0.is_contiguous());
+    TORCH_CHECK(i1.is_contiguous());
+    TORCH_CHECK(i2.is_contiguous());
 
     // If we are passing in query, key, value all the same tensors then we have
     // packed them into one tensor and need to slice for flash attention
     Tensor atten_buffer = at::_flash_scaled_dot_product_attention(
-        qkv_buffer_reshaped.index({at::indexing::Slice(), 0}),
```

Comment:
I just added the `qkv_buffer_reshaped.index({at::indexing::Slice(), 0}).contiguous()` for the next three and that seemed to fix as well. Not sure if that is less perf but felt like fewer dispatch calls ... premature optimization is the way

Reply:
That'll cause one call to `contiguous` for each Tensor, which means 3 kernel launches. I don't think these memcpys take a lot of GPU time (didn't look at a trace), but if we can do it with one call to contiguous I suspect it's more efficient.

Reply:
even more interestingly, at least to me, calling contiguous on the individual q,k,v slices fixes the repro however there are still cuda memory errors for padding patterns. This method of transposing then making the entire packed project contiguous appears to work for the entire sweep.